node-passport-login/models/userModel.js

```diff
@@ -19,6 +19,25 @@ const UserSchema = new mongoose.Schema({
   },
 });
 
+// hashing password
+userSchema.pre('save', async function (next) {
+  // Only run this function if password was actually modified
+  if (!this.isModified('password')) return next();
+
+  // Hash the password with cost of 12
+  this.password = await bcrypt.hash(this.password, 12);
+
+  next();
+});
+
+// checking password
+userSchema.methods.correctPassword = async function (
+  candidatePassword,
+  userPassword
+) {
+  return await bcrypt.compare(candidatePassword, userPassword);
+};
+
```

Comment:
pre-function for hash password and checking password correct or not functionality.